File: web/src/core/feature/exchange/domain/repository/ExchangeRepository.ts
import type {CupExchange} from "../entity/CupExchange";

export interface ExchangeRepository {
    getToday(): Promise<CupExchange>

    getToADay(date: Date): Promise<CupExchange>

    getCachedToday(): Promise<CupExchange | null>

    getToARankOfDays(dateFrom: Date, dateTo: Date): Promise<CupExchange[]>
}

File: app/src/main/java/com/elitec/alejotaller/feature/exchange/domain/repository/ExchangeRepository.kt
package com.elitec.alejotaller.feature.exchange.domain.repository

import com.elitec.alejotaller.feature.exchange.domain.entity.CupExchange

interface ExchangeRepository {
    suspend fun getToday(): CupExchange
    suspend fun getCachedToday(): CupExchange?
}
